src/DotNet.Testcontainers.Tests/Unit/Containers/Unix/Database/CouchbaseTestcontainerTest.cs

```diff
@@ -0,0 +1,157 @@
+namespace DotNet.Testcontainers.Tests.Unit.Containers.Unix.Database
+{
+  using System;
+  using System.Threading.Tasks;
+  using Couchbase;
+  using Testcontainers.Containers.Builders;
+  using Testcontainers.Containers.Configurations.Databases;
+  using Testcontainers.Containers.Modules.Databases;
+  using Xunit;
+
+  public class CouchbaseTestcontainerTest
+  {
+
+    /// <summary>
+    /// Sample Couchbase document class for test
+    /// </summary>
+    private class Customer
+    {
+      public string Name { get; set; }
+      public int Age{ get; set; }
+    }
+
+    [Fact]
+    public async Task ConnectionEstablished_WithCrudTest()
+    {
+      //Given
+      var couchbaseTestcontainer = new TestcontainersBuilder<CouchbaseTestcontainer>()
+        .WithDatabase(new CouchbaseTestcontainerConfiguration("Administrator", "password", "Customer"))
+        .WithExposedPort(8091)
+        .WithExposedPort(11210)
+        .WithPortBinding(8091, 8091)
+        .WithPortBinding(11210, 11210)
+        .Build();
+
+      await couchbaseTestcontainer.StartAsync();
+
+      //When & Then
+      var cluster = await Cluster.ConnectAsync(couchbaseTestcontainer.ConnectionString, "Administrator","password");
+      var bucket = await cluster.BucketAsync("Customer");
+      var collection = bucket.DefaultCollection();
+
+      // Create
+      await collection.InsertAsync(
+        "customer-id-1",
+        new Customer {Name = "Mustafa", Age = 29}
+        );
+
+      // Read
+      var customer = (await collection.GetAsync("customer-id-1")).ContentAs<Customer>();
+      Assert.True(customer.Name == "Mustafa");
+      Assert.True(customer.Age == 29);
+
+      //Update
+      customer.Age = 30;
+      customer.Name = "Onur";
+      await collection.UpsertAsync("customer-id-1", customer);
+
+      var updatedCustomer = (await collection.GetAsync("customer-id-1")).ContentAs<Customer>();
+      Assert.True(updatedCustomer.Name == "Onur");
+      Assert.True(updatedCustomer.Age == 30);
+
+      //Delete
+      await collection.RemoveAsync("customer-id-1");
+      var existsResult = await collection.ExistsAsync("customer-id-1");
+      Assert.False(existsResult.Exists);
+
+      await couchbaseTestcontainer.StopAsync();
+    }
+
+    [Fact]
+    public void ShouldNotThrowsArgumentOutOfRangeException_WhenClusterConfigurationIsValid()
+    {
+      //Given
+      var configuration = new CouchbaseTestcontainerConfiguration("Administrator", "password", "dotnet-testcontainers");
+
+      //When & Then
+      Func<CouchbaseTestcontainerConfiguration> action = () => configuration
```

Comment:
You don't invoke the action and check the results here. I think you should at least call the action. If we switch to getters and setters we can even test the result. Would be nice.

Reply:
You are absolutely right! I will reconsider these tests after update configuration class with your suggestion. Also, I am looking something like Assert.NotThrows , do you have any advice?

Reply:
I think there is:

```
var ex = Record.Exception(() => // Code that does not throw);
Assert.Null(ex);
```